### PR TITLE
Passing IndicesService to ingest processor factory with processor params

### PR DESCRIPTION
**Pull request by martin-gaievski**
### Description
This change solves the problem of ingest pipeline processors not being able to access index mapping information. By passing reference to IngestService we opening this possibility, processor may use it to read mapping like this :

in the ingest plugin
```
    Map<String, Processor.Factory> getProcessors(Processor.Parameters parameters) {
        return Map.of(
            MyIngestProcessor.TYPE,
            new MyIngestProcessorFactory(parameters.indicesService)
        );
    }
```

in the processor factory
```
    public MyIngestProcessor create(... ) throws Exception {
        return new MyIngestProcessor(parameters.indicesService);
    }
```

in the processor
```
    IndicesService indicesService;
    
    void execute(IngestDocument ingestDocument, BiConsumer<IngestDocument, Exception> handler) {
        String index = ingestDocument.getSourceAndMetadata().get(IndexFieldMapper.NAME);
        MappingMetadata mappingMeta = indicesService.clusterService().state().metadata().index(indexName).mapping();
    }

```

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by msfroh** [comment]
As a nitpick, the title should probably say "Passing IndicesService..."

**Comment by msfroh** [comment]
@martin-gaievski  -- Are you still planning to move forward with this one? Looks like we have some merge conflicts to resolve.

**Comment by martin-gaievski** [comment]
Yes, I think the change still makes sense, resolving merge conflicts now

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - GHA to verify checklist items completion in PR descriptions ([#10800](https://github.com/opensearch-project/OpenSearch/pull/10800))
 - [Remote cluster state] Restore cluster state version during remote state auto restore ([#10853](https://github.com/opensearch-project/OpenSearch/pull/10853))
 - Add back half_float BKD based sort query optimization ([#11024](https://github.com/opensearch-project/OpenSearch/pull/11024))
+- Allowing pipeline processors to access index maping info by passing ingest service ref as part of the processor factory parameters ([#10307](https://github.com/opensearch-project/OpenSearch/pull/10307))
```
Comment (msfroh) [inline CHANGELOG.md:25]: Do you want to release this in 3.0? Or do you intend to backport to 2.x?

If you plan to backport, please move this down to the Unreleased 2.x section.

Comment (martin-gaievski) [reply CHANGELOG.md]: Yes, it makes sense to backport that. I've made change to the CHANGELOG, can you please help with the backport label, I don't have permissions to add it myself

File: CHANGELOG.md [hunk 2]
```diff
@@ -103,6 +103,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add cluster state stats ([#10670](https://github.com/opensearch-project/OpenSearch/pull/10670))
 - Adding slf4j license header to LoggerMessageFormat.java ([#11069](https://github.com/opensearch-project/OpenSearch/pull/11069))
 - [Streaming Indexing] Introduce new experimental server HTTP transport based on Netty 4 and Project Reactor (Reactor Netty) ([#9672](https://github.com/opensearch-project/OpenSearch/pull/9672))
+- Allowing pipeline processors to access index maping info by passing ingest service ref as part of the processor factory parameters ([#10307](https://github.com/opensearch-project/OpenSearch/pull/10307))
```
Comment (ryanbogan) [inline CHANGELOG.md:106]: Minor typo: mapping

**Review by ryanbogan — APPROVED** [review]
One small suggestion, everything else looks good to me!

File: CHANGELOG.md (hunk 2 above)
Comment (martin-gaievski) [reply CHANGELOG.md]: ack, fixed issue description

**Review by msfroh — APPROVED** [review]

**Pull request merged**